relay/polkadot/src/lib.rs

```diff
@@ -1780,22 +1841,102 @@ pub type Migrations = (migrations::Unreleased, migrations::Permanent);
 	}
 
+	pub struct GetLegacyLeaseImpl;
+	impl coretime::migration::GetLegacyLease<BlockNumber> for GetLegacyLeaseImpl {
+		fn get_parachain_lease_in_blocks(para: ParaId) -> Option<BlockNumber> {
+			let now = frame_system::Pallet::<Runtime>::block_number();
+			let lease = slots::Leases::<Runtime>::get(para);
+			if lease.is_empty() {
+				return None
+			}
+			let (index, _) =
+				<slots::Pallet<Runtime> as Leaser<BlockNumber>>::lease_period_index(now)?;
+			Some(index.saturating_add(lease.len() as u32).saturating_mul(LeasePeriod::get()))
+		}
+	}
+
+	/// Cancel all ongoing auctions.
+	///
+	/// Any leases that come into existence after coretime was launched will not be served. Yet,
+	/// any ongoing auctions must be cancelled.
+	///
+	/// Safety:
+	///
+	/// - After coretime is launched, there are no auctions anymore. So if this forgotten to
+	/// be removed after the runtime upgrade, running this again on the next one is harmless.
+	/// - I am assuming scheduler `TaskName`s are unique, so removal of the scheduled entry
+	/// multiple times should also be fine.
+	pub struct CancelAuctions;
+	impl OnRuntimeUpgrade for CancelAuctions {
+		fn on_runtime_upgrade() -> Weight {
+			if let Err(err) = Auctions::cancel_auction(frame_system::RawOrigin::Root.into()) {
+				log::debug!(target: "runtime", "Cancelling auctions failed: {:?}", err);
+			}
+			// Cancel scheduled auction as well:
+			if let Err(err) = Scheduler::cancel_named(
+				pallet_custom_origins::Origin::AuctionAdmin.into(),
+				[
+					0x87, 0xa8, 0x71, 0xb4, 0xd6, 0x21, 0xf0, 0xb9, 0x73, 0x47, 0x5a, 0xaf, 0xcc,
+					0x32, 0x61, 0x0b, 0xd7, 0x68, 0x8f, 0x15, 0x02, 0x33, 0x8a, 0xcd, 0x00, 0xee,
+					0x48, 0x8a, 0xc3, 0x62, 0x0f, 0x4c,
+				],
+			) {
+				log::debug!(target: "runtime", "Cancelling scheduled auctions failed: {:?}", err);
+			}
+			use pallet_scheduler::WeightInfo as _;
+			use polkadot_runtime_common::auctions::WeightInfo as _;
+			weights::polkadot_runtime_common_auctions::WeightInfo::<Runtime>::cancel_auction()
+				.saturating_add(weights::pallet_scheduler::WeightInfo::<Runtime>::cancel_named(
+					<Runtime as pallet_scheduler::Config>::MaxScheduledPerBlock::get(),
+				))
+		}
+	}
+
+	// Migrate storage for pallet rename `OnDemandAssignmentProvider` -> `OnDemand`
+	pub struct OnDemandRename;
+	impl OnRuntimeUpgrade for OnDemandRename {
+		fn on_runtime_upgrade() -> Weight {
+			move_storage_from_pallet(b"Pallet", b"OnDemandAssignmentProvider", b"OnDemand");
+			move_storage_from_pallet(b"ParaIdAffinity", b"OnDemandAssignmentProvider", b"OnDemand");
+			move_storage_from_pallet(b"QueueStatus", b"OnDemandAssignmentProvider", b"OnDemand");
+			move_storage_from_pallet(b"FreeEntries", b"OnDemandAssignmentProvider", b"OnDemand");
+			move_storage_from_pallet(
+				b"AffinityEntries",
+				b"OnDemandAssignmentProvider",
+				b"OnDemand",
+			);
+			move_storage_from_pallet(b"Revenue", b"OnDemandAssignmentProvider", b"OnDemand");
```

Comment:
@s0me0ne-unkn0wn does this look right to you?

As far as I understand I need to call `move_storage_from_pallet` for each storage item in the pallet?
Are the pallet names correct?

Ideally I should use `get_storage_prefix()` here but the storage items are not accessible here so I hardcoded them :1st_place_medal:

Reply:
> @s0me0ne-unkn0wn does this look right to you?

Well, it looks totally fine to me, but it should be taken into account that I didn't ever do it myself as well tbh :laughing:

> As far as I understand I need to call `move_storage_from_pallet` for each storage item in the pallet?
> Are the pallet names correct?

My understanding is the same, and the naming seems to be fine. I still hope to get some reviews from guys who have closer experience :slightly_smiling_face: